apollo-router/src/plugins/telemetry/tracing/apollo_telemetry.rs

```diff
@@ -123,7 +122,7 @@ impl Exporter {
         let apollo_exporter =
             ApolloExporter::new(&endpoint, &apollo_key, &apollo_graph_ref, &schema_id)?;
         Ok(Self {
-            spans_by_parent_id: LruCache::new(buffer_size),
+            spans_by_parent_id: HashMap::with_capacity(buffer_size),
```

Comment:
This reuses the same `buffer_size` configuration but for a very different purpose. The parameter for `LruCache::new` sets the upper bound for the cache size. Beyond that it starts evicting entries. `HashMap::with_capacity` influences the size of the initial heap allocation but the map can reallocate to be bigger.

If we’re using `HashMap`, I don’t think making the initial capacity configurable is useful. And if there are cases where the map stays empty we might even be better off with `HashMap::new`.